### PR TITLE
Bug 1815632 - Adds https to non-http(s) protocols when adding new login

### PR DESCRIPTION
**Pull request by gitstart**
Issue link: https://bugzilla.mozilla.org/show_bug.cgi?id=1815632

### Fix details
Adds `https://` at start of `hostname` if user not added `http` or `https`

### Steps to test

1. Open three dots menu from home page
2. Navigate to Settings -> Logins and Passwords -> Saved Logins -> Add Login
3. Enter hostname without https:// or http:// and user name and password
4. Save login

### Screenshots
Issue video:
https://www.loom.com/share/0f0c402f7cd74ff9bf8e8b3e0f7b0795

Fix video:
https://www.loom.com/share/5a941a8d9394489fb6f480ee5fb3a87d

### Pull Request checklist
 - [x] Quality: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
 - [ ] Tests: This PR includes thorough tests or an explanation of why it does not
 - [ ] Changelog: This PR includes [a changelog entry](https://github.com/mozilla-mobile/android-components/blob/main/docs/changelog.md) or does not need one
 - [ ] Accessibility: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/main/android/accessibility_guide.md) or does not include any user facing features

### After merge
- Milestone: Make sure issues closed by this pull request are added to the [milestone](https://github.com/mozilla-mobile/android-components/milestones) of the version currently in development.
- Breaking Changes: If this is a breaking change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

---
This code was written and reviewed by GitStart Community. Growing future engineers, one PR at a time.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1815632

### CODE REVIEW

File: fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt [hunk 1]
```diff
@@ -339,6 +339,15 @@ class AddLoginFragment : Fragment(R.layout.fragment_add_login), MenuProvider {
         activity?.invalidateMenu()
     }
 
+    private fun webProtocolTreatment(hostname: String): String {
```
Comment (t-p-white) [inline fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt:342]: Can this check be moved to URLStringUtils.kt, where we can have something like:
```  
    /**
     * Returns true if the [String] uses the "http" or "https" protocol scheme.
     */
    fun String.isHttpOrHttps() =
        startsWith(HTTPS, ignoreCase = true) || startsWith(HTTP, ignoreCase = true)
```

then `webProtocolTreatment` function and companion object could be removed and even used directly as something like e.g.
```
interactor.onAddLogin(
                with(binding.hostnameText.text.toString()) {
                    if (isHttpOrHttps()) {
                        this
                    } else {
                        "$HTTPS_PROTOCOL$this"
                    }
                },
                binding.usernameText.text.toString(),
                binding.passwordText.text.toString(),
            )
```

You could then write tests for your `isHttpOrHttps` fun

**Review by t-p-white — CHANGES_REQUESTED** [review]
Thank you for your contribution @gitstart, nice work. Please see review comments

File: fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt (hunk 1 above)
Comment (gitstart) [reply fenix/app/src/main/java/org/mozilla/fenix/settings/logins/fragment/AddLoginFragment.kt]: Refactored code and added unit tests.

**Comment by gitstart** [comment]
Hi @t-p-white Thanks for reviewing PR. 

Requested changes has been addressed.
Can you please re-review?

**Comment by t-p-white** [comment]
@gitstart thanks for updating! Curious if there was a reason to not use `URLStringUtils`? 🙂 
My sincere apologies, since re-reviewing I see there is already a `isHttpOrHttps` fun in `URLStringUtils` - preferably we would use this and the only changes required here would be in `AddLoginFragment`. Appreciate the effort you've already put into this 🙏

**Comment by gitstart** [comment]
Thanks @t-p-white for reviewing. Requested changes has been pushed. Can you please re-review?

**Comment by t-p-white** [comment]
bors try

**Comment by t-p-white** [comment]
bors try

**Comment by t-p-white** [comment]
bors try

**Comment by DonalMe** [comment]
@t-p-white this has a `needs landing` label but your approval review was never added. Please take a look.

**Review by t-p-white — APPROVED** [review]
@gitstart lgtm 😎

**Comment by t-p-white** [comment]
@DonalMe 👍

**Pull request merged**